embassy-traits/src/qei.rs

```diff
@@ -0,0 +1,13 @@
+use core::future::Future;
+use core::pin::Pin;
+use embedded_hal::Direction;
+
+pub trait Qei {
+    type RotateFuture<'a>: Future<Output = Direction> + 'a;
+
+    fn wait_for_rotate<'a>(
```

Comment:
Can you add docs?

Reply:
@xoviat PR looks good to me, only thing missing is this. Can you add docs, outlining for example:
- What does the function do
- What are the units? rotations, degrees, 
- When does it return
etc